plugins/modules/consul_agent_check.py

```diff
@@ -0,0 +1,270 @@
+    interval: 60s
+    tcp: localhost:80
+    notes: "Nginx Check"
+
+- name: register http check for service 'nginx'
+  community.general.consul_agent_check:
+    name: nginx_http_check
+    service_id: nginx
+    interval: 60s
+    http: http://localhost:80/status
+    notes: "Nginx Check"
+    
+
+- name: Remove check for service 'nginx'
+  community.general.consul_agent_check:
+    state: absent
+    id: nginx_http_check
+    service_id: "{{ nginx_service.ID }}"
+
+
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.community.general.plugins.module_utils.consul import (
+    AUTH_ARGUMENTS_SPEC,
+    OPERATION_CREATE,
+    OPERATION_UPDATE,
+    OPERATION_DELETE,
+    OPERATION_READ,
+    _ConsulModule,
+    camel_case_key,
+    RequestError,
+    validate_check,
+)
+
+_ARGUMENT_SPEC = {
+    "state": dict(default="present", choices=["present", "absent"]),
+    "name": dict(type='str'),
+    "id": dict(type='str'),
+    "interval": dict(type='str'),
+    "notes": dict(type='str'),
+    "http": dict(type='str'),
+    "tcp": dict(type='str'),
+    "ttl": dict(type='str'),
+    "timeout": dict(type='str'),
+    "service_id": dict(type='str'),
+}
+
+_MUTUALLY_EXCLUSIVE = [
+    ('args', 'ttl', 'tcp', 'http'),
+]
+
+_REQUIRED_IF = [
+    ('state', 'present', ['name']),
+    ('state', 'absent', ['id']),
+]
+
+_REQUIRED_BY = {
+    'args': 'interval',
+    'http': 'interval',
+    'tcp': 'interval',
+}
+
+_ARGUMENT_SPEC.update(AUTH_ARGUMENTS_SPEC)
+
+
+class ConsulAgentCheckModule(_ConsulModule):
+    api_endpoint = "agent/check"
+    result_key = "check"
+    unique_identifier = "id"
+
+    def endpoint_url(self, operation, identifier=None):
+        if operation == OPERATION_READ:
+            return "/".join([self.api_endpoint + 's'])
+        if operation in [OPERATION_CREATE, OPERATION_UPDATE]:
+            return "/".join([self.api_endpoint, "register"])
+        if operation == OPERATION_DELETE:
+            return "/".join([self.api_endpoint, "deregister", identifier])
+
+        return super(ConsulAgentCheckModule, self).endpoint_url(operation, identifier)
+
+    def read_object(self):
+        url = self.endpoint_url(OPERATION_READ)
+        checks = self.get(url)
+
+        if self.params.get(self.unique_identifier) in checks:
+            return checks[self.params.get(self.unique_identifier)]
+        else:
+            return
+
+    def prepare_object(self, existing, obj):
+        existing = super(ConsulAgentCheckModule, self).prepare_object(existing, obj)
+
+        operational_attributes = {"Node", "CheckID", "Output", "ServiceName", "ServiceTags",
+                                  "Status", "Type", "ExposedPort", "Definition"}
+        existing = {
+            k: v for k, v in existing.items() if k not in operational_attributes
+        }
+
+        validate_check(existing)
+
+        if 'Args' in existing and existing['Args']:
+            existing['Args'] = ["sh", "-c", existing['Args']]
```

Comment:
I do not really understand why we would want to add `"sh", "-c"` here. Forcing checks to execute through a shell is overkill

Reply:
I looked in the documentation, and they said it should be executed with "sh", "-c".  https://developer.hashicorp.com/consul/api-docs/agent/check#args
I also had a look at the python-consul package. They also do similar wrapping. 
https://github.com/python-consul/python-consul/blob/master/consul/base.py#L62

Reply:
The docs don't say you should be executing with `sh -c`. The docs say that if you want the old `Script` behavior you should run with `sh -c`.  The docs you linked explicitly state:
> In Consul 1.0, the Args array was added so that checks can be run without a shell.

The whole point of `Args` is to be able to run without a shell ;)

Reply:
The "old" python-consul probably tried to emulate the now deprecated and removed `Script` behavior. We don't need/want to support anything older than consul 1.0.

Reply:
Ah ok. Then I misinterpreted it :) I will remove it.

Reply:
Yeah, just for clarification. In early Consul versions you could set:
`Script: my_test.sh`
and it internally would basically execute: `sh -c my_test.sh`.

In 1.0 the introduced `Args` and at some point removed `Script`. Now when you set:
`Args: [my_test.sh]` it will literally execute `my_test.sh` as check. To get back to the old `Script` behavior you can now set `Args: [sh, -c, my_test.sh]` which might (?) be useful if you are using something like
```
for i in `ls …`; do check; done
```
in your check. But generally when you actually execute a binary it is not needed.